packages/mobile/src/components/Avatar.tsx

```diff
@@ -49,20 +45,51 @@ export function getE164Number(e164Number?: string, recipient?: Recipient) {
 }
 
 export function Avatar(props: Props) {
-  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
-  const { recipient, e164Number, iconSize = DEFAULT_ICON_SIZE, displayNameStyle } = props
+  const defaultCountryCode = useSelector(defaultCountryCodeSelector) ?? undefined
+  const { address, recipient, e164Number, iconSize = DEFAULT_ICON_SIZE, displayNameStyle } = props
+
+  const name = getDisplayName(props)
+  const e164NumberToShow = getE164Number(e164Number, recipient)
+  const thumbnailPath = getRecipientThumbnail(recipient)
 
   return (
-    <BaseAvatar
-      {...props}
-      defaultCountryCode={defaultCountryCode ? defaultCountryCode : undefined}
-      name={getDisplayName(props)}
-      e164Number={getE164Number(e164Number, recipient)}
-      iconSize={iconSize}
-      thumbnailPath={getRecipientThumbnail(recipient)}
-      displayNameStyle={displayNameStyle}
-    />
+    <View style={styles.container}>
+      <ContactCircle thumbnailPath={thumbnailPath} name={name} address={address} size={iconSize} />
+      <Text
+        style={[displayNameStyle || fontStyles.small500, styles.contactName]}
+        numberOfLines={1}
+        ellipsizeMode="tail"
+      >
+        {name}
+      </Text>
+      {e164NumberToShow ? (
+        <PhoneNumberWithFlag
+          e164PhoneNumber={e164NumberToShow}
+          defaultCountryCode={defaultCountryCode}
+        />
+      ) : null}
+
+      {!e164Number && address ? (
+        <Text style={[fontStyles.small, styles.contactName]} numberOfLines={1} ellipsizeMode="tail">
+          {'#' + address}
```

Comment:
Probably, this was this way from before: https://github.com/celo-org/celo-monorepo/pull/6457/files#diff-35c5c949f9f4c5c4d103094c21a62c052eb60363c234870ea8f1f00f6c5ee48aL61, I moved this file from the `react-components` to the `mobile` package because it used `ContactCircle` and I moved that one because I needed access to the Redux store and the `DefaultAvatar`.

I think it makes sense to change though, I'll do that